### ISSUE TITLE
Ability to set P2 mirror for Maven plugin

### ISSUE THREAD

**Issue by jglick** [diffplug/spotless#1697]
I have been getting sporadic errors like

```
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.36.0:check (default) on project …: Execution default of goal com.diffplug.spotless:spotless-maven-plugin:2.36.0:check failed: 502 at https://groovy.jfrog.io/artifactory/plugins-release/org/codehaus/groovy/groovy-eclipse-integration/4.8.0/e4.26/plugins/org.codehaus.groovy_4.0.7.v202212311456-e2212-RELEASE.jar -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.36.0:check (default) on project …: Execution default of goal com.diffplug.spotless:spotless-maven-plugin:2.36.0:check failed: 502 at https://groovy.jfrog.io/artifactory/plugins-release/org/codehaus/groovy/groovy-eclipse-integration/4.8.0/e4.26/plugins/org.codehaus.groovy_4.0.7.v202212311456-e2212-RELEASE.jar
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:347)
    at …
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.diffplug.spotless:spotless-maven-plugin:2.36.0:check failed: 502 at https://groovy.jfrog.io/artifactory/plugins-release/org/codehaus/groovy/groovy-eclipse-integration/4.8.0/e4.26/plugins/org.codehaus.groovy_4.0.7.v202212311456-e2212-RELEASE.jar
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
    at …
Caused by: java.lang.IllegalArgumentException: 502 at https://groovy.jfrog.io/artifactory/plugins-release/org/codehaus/groovy/groovy-eclipse-integration/4.8.0/e4.26/plugins/org.codehaus.groovy_4.0.7.v202212311456-e2212-RELEASE.jar
    at dev.equo.solstice.p2.JarCache.download (JarCache.java:52)
    at dev.equo.solstice.p2.P2Client.download (P2Client.java:69)
    at dev.equo.solstice.p2.P2QueryResult.<init> (P2QueryResult.java:35)
    at dev.equo.solstice.p2.P2Model.query (P2Model.java:126)
    at com.diffplug.spotless.extra.EquoBasedStepBuilder.get (EquoBasedStepBuilder.java:103)
    at com.diffplug.spotless.FormatterStepImpl.calculateState (FormatterStepImpl.java:58)
    at com.diffplug.spotless.LazyForwardingEquality.state (LazyForwardingEquality.java:56)
    at com.diffplug.spotless.LazyForwardingEquality.toBytes (LazyForwardingEquality.java:85)
    at com.diffplug.spotless.LazyForwardingEquality.hashCode (LazyForwardingEquality.java:102)
    at java.util.ArrayList.hashCodeRange (ArrayList.java:627)
    at java.util.ArrayList.hashCode (ArrayList.java:614)
    at com.diffplug.spotless.Formatter.hashCode (Formatter.java:276)
    at java.util.HashMap.hash (HashMap.java:340)
    at java.util.HashMap.put (HashMap.java:608)
    at com.diffplug.spotless.maven.FormattersHolder.create (FormattersHolder.java:43)
    at com.diffplug.spotless.maven.AbstractSpotlessMojo.execute (AbstractSpotlessMojo.java:224)
    at …
```

This despite my `~/.m2/settings.xml` defining a `mirrorOf` pointing to a Nexus pull-through proxy of Maven Central, violating the general expectation of Maven plugins.

If this artifact, whatever it is, cannot be hosted on Central, then is there at least some way to define a separate mirror for it so CI builds are not at the mercy of an external server?

#1635 seems to only support Gradle. #1669 sounds related but then discusses Gradle as well. #1658 sounds related but distinct—about a generic HTTP proxy, rather than a mirror.

**Comment by jtnord** [comment]
it should be noted that the configuration of spotless here is only using 

```
      <plugin>
        <groupId>com.diffplug.spotless</groupId>
        <artifactId>spotless-maven-plugin</artifactId>
        <configuration>
          <java>
            <endWithNewline />
            <importOrder />
            <indent>
              <spaces>true</spaces>
            </indent>
            <palantirJavaFormat />
            <removeUnusedImports />
            <trimTrailingWhitespace />
          </java>
          <pom>
            <sortPom>
              <expandEmptyElements>false</expandEmptyElements>
              <sortDependencies>scope,groupId,artifactId</sortDependencies>
              <sortDependencyExclusions>groupId,artifactId</sortDependencyExclusions>
              <spaceBeforeCloseEmptyElement>true</spaceBeforeCloseEmptyElement>
            </sortPom>
          </pom>
        </configuration>
```

so it is not even clear why non configured formatters are even configured/run.

**Comment by nedtwigg** [comment]
Yeah, eclipse.org was down with 502 errors ~two weeks ago, so that's what was happening.

- #1635 makes p2 mirroring work, and then adds it to the Gradle DSL.
- You are correct that it has not been exposed in the Maven DSL, a PR for that is welcome.

**Comment by nedtwigg** [comment]
Shipped in `plugin-gradle 6.22.0` and `plugin-maven 2.40.0`.
